src/plugins/validation/2and3/semantic-validators/parameters-ibm.js

```diff
@@ -157,6 +157,40 @@ module.exports.validate = function({ jsSpec, isOAS3 }, config) {
   return { errors: result.error, warnings: result.warning };
 };
 
+function isParameter(path, isOAS3) {
+  const pathsForParameters = [
+    'get',
+    'put',
+    'post',
+    'delete',
+    'options',
+    'head',
+    'patch',
+    'trace',
+    'components'
+  ];
+
+  const pathCheck =
+    pathsForParameters.includes(path[path.length - 3]) ||
+    (path[path.length - 4] === 'paths' &&
+      path[2] === 'parameters' &&
+      path.length === 4);
+
+  const isTopLevelParameter =
+    !isOAS3 && path[0] === 'parameters' && path.length === 2;
+
+  const parameterObj = path[path.length - 2] === 'parameters';
+
+  if (parameterObj) {
+    if (isOAS3) {
+      return pathCheck;
+    } else if (!isOAS3) {
+      //swagger 2 allows parameters to be in the top level, this checks for that
+      return pathCheck || isTopLevelParameter;
+    }
+  }
+}
```

Comment:
I think this is still too complicated. Since this is tricky, I'll write some code as a suggestion:
```js
function isParameter(path, isOAS3) {
  const pathsForParameters = [
    'get',
    'put',
    'post',
    'delete',
    'options',
    'head',
    'patch',
    'trace',
    'components'
  ];
  
  const inParametersSection = path[path.length - 2] === 'parameters';

  // the above check is a necessary but not sufficient check for a parameter object
  // use the following checks to verify the object is where a parameter is supposed to be.
  // without these, a schema property named "parameters" would get validated as a parameter
  const isParameterByPath = pathsForParameters.includes(path[path.length - 3]);
  const isPathItemParameter =
    path[path.length - 4] === 'paths' && path.length === 4;
  const isTopLevelParameter =
    !isOAS3 && path[0] === 'parameters' && path.length === 2;

  return inParametersSection &&
    (isParameterByPath || isPathItemParameter || isTopLevelParameter);
}
```